catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/converter/mgmp/MgmpConverter.java

```diff
@@ -0,0 +1,1230 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.alliance.catalog.converter.mgmp;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.alliance.catalog.core.api.types.Isr;
+import org.codice.alliance.catalog.core.api.types.Security;
+import org.codice.alliance.catalog.transformer.mgmp.MgmpConstants;
+import org.codice.ddf.spatial.ogc.csw.catalog.common.GmdConstants;
+import org.codice.ddf.spatial.ogc.csw.catalog.converter.AbstractGmdConverter;
+import org.codice.ddf.spatial.ogc.csw.catalog.converter.XstreamPathValueTracker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.thoughtworks.xstream.io.path.Path;
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.io.ParseException;
+import com.vividsolutions.jts.io.WKTReader;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Associations;
+import ddf.catalog.data.types.Contact;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.data.types.DateTime;
+import ddf.catalog.data.types.Location;
+import ddf.catalog.data.types.Media;
+import ddf.catalog.data.types.Topic;
+
+/**
+ * Not thread safe.
+ */
+public class MgmpConverter extends AbstractGmdConverter {
```

Comment:
Wish we could extract some of this code to external classes. It would reduce the class' size and might make it easier to understand and unit test. Thoughts?